seed/seedwriter/writer.go

```diff
@@ -427,9 +427,23 @@ func (w *Writer) SetOptionsSnaps(optSnaps []*OptionsSnap) error {
 			if err != nil {
 				return fmt.Errorf("cannot use option channel for snap %q: %v", whichSnap, err)
 			}
-			if err := w.policy.checkSnapChannel(ch, whichSnap); err != nil {
```

Comment:
AFAIU the original check is intentional and if you are using a mode with grade > dangerous, the channel is already specified in the model itself, thus it is a mistake to attempt to pass one in the command line.

Note that it is enough to pass the snap name as `--snap <name>` to include an optional snap. During seeding the snap will be set up to track the right channel